compiler/src/iree/compiler/Codegen/LLVMGPU/test/transpose_pipeline_test.mlir

```diff
@@ -105,3 +105,207 @@ module attributes {hal.device.targets = [#device_target_cuda]} {
+          ^bb0(%arg0: f32, %arg1: f32, %arg2: f32):
+            %7 = arith.addf %arg0, %arg1 : f32
+            linalg.yield %7 : f32
+          } -> tensor<768x2048x1024xf32>
+          flow.dispatch.tensor.store %6, %2, offsets = [0, 0, 0], sizes = [768, 2048, 1024], strides = [1, 1, 1] : tensor<768x2048x1024xf32> -> !flow.dispatch.tensor<writeonly:768x2048x1024xf32>
+          return
+        }
+      }
+    }
+  }
+}
+
+// CHECK-LABEL:   hal.executable public @transpose_3d_no_dispatch_0_generic_768x2048x1024 {
+//       CHECK:   %[[CST:.*]] = arith.constant 0.000000e+00 : f32
+//       CHECK:   %[[C64:.*]] = arith.constant 64 : index
+//       CHECK:   %[[C0:.*]] = arith.constant 0 : index
+//       CHECK:   %[[D0:.*]] = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) offset(%[[C0]]) alignment(64) : memref<2048x768x1024xf32>
+//       CHECK:   memref.assume_alignment %[[D0]], 64 : memref<2048x768x1024xf32>
+//       CHECK:   %[[D1:.*]] = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) offset(%[[C0]]) alignment(64) : memref<768x2048x1024xf32>
+//       CHECK:   memref.assume_alignment %[[D1]], 64 : memref<768x2048x1024xf32>
+//       CHECK:   %[[D2:.*]] = hal.interface.binding.subspan set(0) binding(2) type(storage_buffer) offset(%[[C0]]) alignment(64) : memref<768x2048x1024xf32>
+//       CHECK:   memref.assume_alignment %[[D2]], 64 : memref<768x2048x1024xf32>
+//       CHECK:   %[[D4:.*]] = memref.subview %[[D2]]{{\[}}%{{.*}}, %{{.*}}, %{{.*}}] [1, 1, 256] [1, 1, 1] : memref<768x2048x1024xf32> to memref<1x1x256xf32, {{.*}}>
+//       CHECK:   %[[D5:.*]] = memref.subview %[[D0]]{{\[}}%{{.*}}, %{{.*}}, %{{.*}}] [1, 1, 256] [1, 1, 1] : memref<2048x768x1024xf32> to memref<1x1x256xf32, {{.*}}>
+//       CHECK:   %[[D6:.*]] = memref.subview %[[D1]]{{\[}}%{{.*}}, %{{.*}}, %{{.*}}] [1, 1, 256] [1, 1, 1] : memref<768x2048x1024xf32> to memref<1x1x256xf32, {{.*}}>
+//       CHECK:   %[[D7:.*]] = gpu.thread_id  x
+//       CHECK:   %[[D8:.*]] = gpu.thread_id  y
+//       CHECK:   %[[D9:.*]] = gpu.thread_id  z
+//       CHECK:   %[[D10:.*]] = affine.apply #map4(%[[D7]])
+//       CHECK:   %[[D11:.*]] = memref.subview %[[D4]][0, 0, %[[D10]]] [1, 1, 4] [1, 1, 1] : memref<1x1x256xf32, #{{.*}}> to memref<1x1x4xf32, #{{.*}}>
+//       CHECK:   %[[D12:.*]] = vector.transfer_read %[[D5]]{{\[}}%[[C0]], %[[C0]], %[[D10]]], %[[CST]] {in_bounds = [true]} : memref<1x1x256xf32, #{{.*}}>, vector<4xf32>
+//       CHECK:   %[[D13:.*]] = vector.transfer_read %[[D6]]{{\[}}%[[C0]], %[[C0]], %[[D10]]], %[[CST]] {in_bounds = [true]} : memref<1x1x256xf32, #{{.*}}>, vector<4xf32>
+//       CHECK:   %[[D14:.*]] = arith.addf %[[D12]], %[[D13]] : vector<4xf32>
+//       CHECK:   vector.transfer_write %[[D14]], %[[D11]]{{\[}}%[[C0]], %[[C0]], %[[C0]]] {in_bounds = [true]} : vector<4xf32>, memref<1x1x4xf32, #{{.*}}>
+//   CHECK-NOT:   gpu.barrier
```

Comment:
Sine this is a negative test I think it is enough to just check that there is no `memref.alloc`